Update test questions and clarify instructions

martynamaron opened the pull request:
I have:
1. Added "passing all tests" to the end goal
2. Made all question instructions clearer
3. Removed Q7 (will be covered in a separate test for JS2) and Q10 (unclear and too complicated)
4. Removed the loop suggestion from Q4, as we encourage using .map() and .forEach()

martynamaron committed:
Make the instructions and end goal clearer

martynamaron committed:
Simplify and clarify test structure

Clarify all question instructions, uncomment function names for
visibility, remove unnecessary questions

alismith-dev commented on module1.js:
```diff
@@ -23,179 +23,106 @@ function question1(name1) {
 
-  @param none
-  @returns [array 0 ... 100] 
+  Use the .forEach() or .map() functions to do this.
```
Hmm, not that I look at this again, `.filter` would be the best choice. Not sure how to word this without directly mentioning that. Maybe that's not a bad thing?

textbook replied:
We could refer to "iteration methods" in general: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#Iteration_methods

martynamaron replied:
Done

alismith-dev commented on module1.js:
```diff
@@ -23,179 +23,106 @@ function question1(name1) {
-*/
+  Write a function that compares two input integers.
 
-// function question2(number1, number2) {
```
If we're uncommenting the question functions we might as well uncomment them in `module.exports` too. The tests should still fail (as all of the functions are no-ops)

martynamaron replied:
Done - I removed the instruction to uncomment the module.exports as well

alismith-dev commented on module1.js:
```diff
@@ -23,179 +23,106 @@ function question1(name1) {
-  +++++++++++++++++++++++++++++++++++++++++++
   Question 5:
-  Write a function called multiply that multiplies two numbers together. It should take two numbers as parameters and return the answer.
+  Write a function that multiplies two input integers or two input floats (positive or negative).
```
I think it might be confusing to mention integers and floats. In terms of this question there's no real difference

martynamaron replied:
Done.
I also realised that the test tries to multiply `0.7` and `2.3`, if you just do `0.7 * 2.3` the result is 1.6099999 - and the test expects `1.61` - do you think that could be a problem?

alismith-dev replied:
Yay for JavaScript floats! 🎉 I can only imagine that was intentional to test for knowledge about this. However I don't think we covered that in the syllabus (you probably know better than I do), so I'd be inclined to remove it. I don't necessarily think it's core to JS knowledge either.

alismith-dev commented:
Looks great!

Commit:
Fix test instructions

martynamaron commented:
I've also fixed some of the instructions to not include "uncommenting functions and module.exports" and added forking and cloning the repo to the readme

Commit:
Fix readme instructions

alismith-dev commented:
I think if we remove that test about floats, this is good to go! Thanks a lot :)

martynamaron committed:
Remove multiplying floats test

alismith-dev commented:
Merging. Thanks :) And congrats on your first github PR 🎆 🍰
